subprojects/core/src/main/java/org/gradle/api/internal/plugins/DefaultObjectConfigurationAction.java

```diff
@@ -97,9 +102,14 @@ private void applyScript(Object script) {
         ScriptSource scriptSource = new UriScriptSource("script", scriptUri);
         ClassLoaderScope classLoaderScopeChild = classLoaderScope.createChild("script-" + scriptUri.toString());
         ScriptHandler scriptHandler = scriptHandlerFactory.create(scriptSource, classLoaderScopeChild);
-        ScriptPlugin configurer = configurerFactory.create(scriptSource, scriptHandler, classLoaderScopeChild, classLoaderScope, false);
-        for (Object target : targets) {
-            configurer.apply(target);
+        final ScriptPlugin configurer = configurerFactory.create(scriptSource, scriptHandler, classLoaderScopeChild, classLoaderScope, false);
```

Comment:
Shouldn't the call to `create` also be part of the build operation?

In the case of `gradle-script-kotlin` that would account for the time spent resolving and instantiating service dependencies.